src/main/java/io/fabric8/maven/XMLFormat.java

```diff
@@ -0,0 +1,105 @@
+package io.fabric8.maven;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringWriter;
+import java.io.UncheckedIOException;
+
+import org.jdom2.Document;
+import org.jdom2.JDOMException;
+import org.jdom2.input.SAXBuilder;
+import org.jdom2.output.Format;
+import org.jdom2.output.XMLOutputter;
+
+/**
+ * Output format for an XML document
+ * <p>
+ * This experimental class is a simple wrapper around JDOM2's {@link XMLOutputter} to provide a simple way to format XML
+ */
+public class XMLFormat {
+
+    private final int indent;
+
+    private final boolean insertLineBreakBetweenMajorSections;
+
+    private XMLFormat(Builder builder) {
+        this.indent = builder.indent;
+        this.insertLineBreakBetweenMajorSections = builder.insertLineBreakBetweenMajorSections;
+    }
+
+    /**
+     * @return the number of spaces to use for indentation
+     */
+    public int getIndent() {
+        return indent;
+    }
+
+    /**
+     * @return true if a line break should be inserted between major sections
+     */
+    public boolean isInsertLineBreakBetweenMajorSections() {
+        return insertLineBreakBetweenMajorSections;
+    }
+
+    /**
+     * Format the XML from the given reader
+     *
+     * @param reader the reader
+     * @return the formatted XML
+     */
+    public String format(Reader reader) {
```

Comment:
So we need to write the output and then read it to format it again?

This would only be used at initial creation?

Typically, I'm not sure why you indent in a second pass while you could indent things properly right away.

And in that case, maybe the new line part should be in Quarkus proper.

Reply:
(btw sorry if my questions are dumb as I don't know the rest of the project!)

Reply:
Starting from version 31, we're no longer changing the format in the target XML. 

Since formatting is needed in some use cases (like when invoked in `CreateProjectMojo`), it's better to have a separate class for that, hence why I came up with this one

Reply:
I'll play around with a `Maven.writeModel` taking a `XMLFormat` as parameter, which may work too, let me investigate further

Reply:
Alright, I included a new method in the `Maven` class to write the model with the predefined format.